### PR TITLE
ROC-3986 remove other toggle features

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: src/main/features/response/components/payment-intention/payment-option.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export class PaymentOptionPage {
 
   private renderView (form: Form<DefendantPaymentOption>, res: express.Response) {
     function isApplicableFor (draft: ResponseDraft): boolean {
-      if (!FeatureToggles.isEnabled('statementOfMeans')) {
+      if (!FeatureToggles.hasAnyAuthorisedFeature(res.locals.claim.features, 'admissions')) {
```
Comment (timja) [inline src/main/features/response/components/payment-intention/payment-option.ts]: extract claim to a var for type safety

File: src/main/features/response/guards/statementOfMeansStateGuard.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export class StatementOfMeansStateGuard {
     return GuardFactory.create((res: express.Response) => {
       const draft: Draft<ResponseDraft> = res.locals.responseDraft
 
-      return StatementOfMeansFeature.isApplicableFor(draft.document)
+      return StatementOfMeansFeature.isApplicableFor(res.locals.claim.features, draft.document)
```
Comment (timja) [inline src/main/features/response/guards/statementOfMeansStateGuard.ts]: extract here too?

File: src/main/features/response/components/payment-intention/payment-option.ts [hunk 3]
```diff
@@ -72,16 +73,17 @@ export class PaymentOptionPage {
...
   private renderView (form: Form<DefendantPaymentOption>, res: express.Response) {
+    const claim: Claim = res.locals.claim
+
     function isApplicableFor (draft: ResponseDraft): boolean {
-      if (!FeatureToggles.isEnabled('statementOfMeans')) {
+      if (!FeatureToggles.hasAnyAuthorisedFeature(claim.features, 'admissions')) {
         return false
       }
-      return draft.isResponseFullyAdmitted()
+      return draft.isResponseFullyAdmitted(claim.features)
```
Comment (geek-soft) [inline src/main/features/response/components/payment-intention/payment-option.ts]: I do not like the fact that now we have to pass features to draft methods. IMO we should separate logic of checking whether feature is enabled from logic checking state of the draft.

I would suggest:

```
# new helper method in separate file
areAdmissionsEnabled(claim: Claim): boolean {
  return FeatureToggles.hasAnyAuthorisedFeature(claim.features, 'admissions')
}

// in many places
ClaimFeatureToggles.areAdmissionsEnabled(claim) && draft.isResponseFullyAdmitted()

// or is other places
if (ClaimFeatureToggles.areAdmissionsEnabled(claim)) {
  if (draft.isResponseFullyAdmitted()) {
    // some logic
  }
}
```

File: src/main/features/testing-support/routes/create-claim-draft.ts [hunk 4]
```diff
@@ -29,6 +32,8 @@ export default express.Router()
       draft.document = new DraftClaim().deserialize(prepareClaimDraft(user.email))
       await new DraftService().save(draft, user.bearerToken)
 
+      await claimStoreClient.addRoleToUser(user, 'cmc-new-features-consent-given')
```
Comment (timja) [inline src/main/features/testing-support/routes/create-claim-draft.ts]: This doesn't check if the user has already given consent or withheld their consent?
I.e. a user who says no to consent should still be able to use this endpoint and not be opted in

File: src/main/features/testing-support/routes/create-claim-draft.ts [hunk 5]
```diff
@@ -32,7 +32,11 @@ export default express.Router()
       draft.document = new DraftClaim().deserialize(prepareClaimDraft(user.email))
       await new DraftService().save(draft, user.bearerToken)
 
-      await claimStoreClient.addRoleToUser(user, 'cmc-new-features-consent-given')
+      const userRoles: string[] = await claimStoreClient.retrieveUserRoles(user)
+
+      if (userRoles.length === 0 || !userRoles.includes('cmc-new-features-consent-given')) {
```
Comment (timja) [inline src/main/features/testing-support/routes/create-claim-draft.ts]: this check isn't right, as it doesn't stop you adding the role if its consent not given
```
!roles.some(role => role.includes('cmc-new-features-consent'))
```

**Pull request merged**